lib/large_object_store.rb

```diff
@@ -68,8 +80,13 @@ def read(key)
         pages
       end
 
-      if data.getbyte(0) == 0x78 && [0x01,0x9C,0xDA].include?(data.getbyte(1))
-        data = Zlib::Inflate.inflate(data)
+      if data[0] == 'z'
+        data = Zlib::Inflate.inflate(data[1..-1])
+      elsif data[0] == '0'
+        data = data[1..-1]
+      else
+        Rails.logger.error "Corrupt large_object_store key #{key}"
```

Comment:
we don't have the logger ... just raise ... this should never happen and if it happens it's a bug that needs to be fixed ?